### ISSUE TITLE
Can't install ripgrep_all with cargo from crates

### ISSUE THREAD

**Issue by halostatue** [phiresky/ripgrep-all#67]
I tried to update `ripgrep_all`, but it failed on what appears to be a dependency issue. If this has been resolved on `master`, it’s probably worth cutting a new version to crates.io that fixes this.

```sh
$ cargo install ripgrep_all
    Updating crates.io index
  Installing ripgrep_all v0.9.6
error: failed to compile `ripgrep_all v0.9.6`, intermediate artifacts can be found at `/var/folders/6t/2xp3hp2j1vd5yyklq0b6_lbw0000gn/T/cargo-install3jDwt5`

Caused by:
  failed to select a version for the requirement `cachedir = "^0.1.1"`
  candidate versions found which didn't match: 0.2.0
  location searched: crates.io index
required by package `ripgrep_all v0.9.6`

$ rustc --version
rustc 1.44.1 (c7087fe00 2020-06-17)
$ cargo --version
cargo 1.44.1 (88ba85757 2020-06-11)
```

**Comment by ramifications123** [comment]
Agree, has anyone found a fix for it?

**Comment by tan-wei** [comment]
Same issue with nightly.

**Comment by wonderfulspam** [comment]
For anyone wondering, this is caused by `cachedir` v0.1.1 being yanked from crates.io. The dependency was removed in [this commit](https://github.com/phiresky/ripgrep-all/commit/963524bbf5ec861cc1d9d2b57e119eb60125751a#diff-80398c5faae3c069e4e6aa2ed11b28c0) which also marked the switch to depending on the nightly compiler. Some of the compiler warnings seem to be caused by the master branch being in a transitional state (eg. unused imports) while another is due to the usage of an unstable nightly feature.

At the time of writing, the master branch depends on two nightly-only features, at least one of which (specialization) is probably a ways off of stabilizing, so it could be a while before this crate can be installed using the stable toolchain. Provided the project continues to rely on those features, of course.

TL;DR: Installing from crates.io will not work until a new version is published. Installing via git using the nightly compiler is the only option for now.

Bonus info: it seems like Cargo's error message [will soon be more explicit](https://github.com/rust-lang/cargo/pull/8565) when you try to install a crate that depends on a yanked crate.

**Comment by blueray453** [comment]
If you use `cargo +nightly install --git https://github.com/phiresky/ripgrep-all.git` then you are basically using master branch which can create some problem. For example you can not use pandoc adapter. https://github.com/phiresky/ripgrep-all/issues/111#issuecomment-882087649

**Comment by blueray453** [comment]
I did some research. https://doc.rust-lang.org/cargo/commands/cargo-install.html#description says that

> By default, the Cargo.lock file that is included with the package will be ignored. This means that Cargo will recompute which versions of dependencies to use, possibly using newer versions that have been released since the package was published. **The --locked flag can be used to force Cargo to use the packaged Cargo.lock file if it is available. This may be useful for ensuring reproducible builds, to use the exact same set of dependencies that were available when the package was published. It may also be useful if a newer version of a dependency is published that no longer builds on your system, or has other problems.** The downside to using --locked is that you will not receive any fixes or updates to any dependency. Note that Cargo did not start publishing Cargo.lock files until version 1.37, which means packages published with prior versions will not have a Cargo.lock file available.

And also https://stackoverflow.com/questions/68509895/what-does-package-foo-in-cargo-lock-is-yanked-in-registry-mean . So, the conclusion for the time being seems to be using:

`cargo install --locked ripgrep_all`

**Comment by Volker-Weissmann** [comment]
> @blueray453 thx, I meet the same problem and use `cargo install --locked ripgrep_all` to install it and success.

Same here

**Comment by donutbrew** [comment]
None of the options here seem to work for me. 

Using the nightly build (or master from git) I get a bunch of compilation errors--things like mismatched types.

using `--locked`

```
$ cargo install --locked ripgrep_all
    Updating crates.io index
warning: no Cargo.lock file published in ripgrep_all v0.9.6
  Installing ripgrep_all v0.9.6
error: failed to compile `ripgrep_all v0.9.6`, intermediate artifacts can be found at `/tmp/cargo-installKe4sL6`

Caused by:
  failed to select a version for the requirement `cachedir = "^0.1.1"`
  candidate versions found which didn't match: 0.3.0, 0.2.0
  location searched: crates.io index
  required by package `ripgrep_all v0.9.6`

```

using `+nightly`:

```
$ cargo +nightly install --locked ripgrep_all
    Updating crates.io index
warning: no Cargo.lock file published in ripgrep_all v0.9.6
  Installing ripgrep_all v0.9.6
error: failed to compile `ripgrep_all v0.9.6`, intermediate artifacts can be found at `/tmp/cargo-installU0KjlM`

Caused by:
  failed to select a version for the requirement `cachedir = "^0.1.1"`
  candidate versions found which didn't match: 0.3.0, 0.2.0
  location searched: crates.io index
  required by package `ripgrep_all v0.9.6`
```

**Comment by danberindei** [comment]
Unfortunately the latest commit has a compilation error, but I was able to install from the Oct 26 commit that upgraded the dependencies:

```
cargo install --git https://github.com/phiresky/ripgrep-all.git --rev 94f06fba37c309d21f3bede7719de660290b7802
```

**Comment by phiresky** [comment]
Master does not contain a working version regardless of if it compiles or not. The last working code is 0.9.6
